docs/standard/serialization/system-text-json-migrate-from-newtonsoft-how-to.md

```diff
@@ -0,0 +1,518 @@
+
+## Comments
+
+During deserialization, Newtonsoft ignores comments in the JSON by default. The `System.Text.Json` default is to throw exceptions for comments because the [RFC 8259](https://tools.ietf.org/html/rfc8259) specification doesn't include them. For information about how to allow comments, see [Allow comments and trailing commas](system-text-json-how-to.md#allow-comments-and-trailing-commas).
+
+## Trailing commas
+
+During deserialization, Newtonsoft ignores trailing commas by default. In some cases, it ignores multiple trailing commas (for example, `[{"Color":"Red"},{"Color":"Green"},,]`). The `System.Text.Json` default is to throw exceptions for trailing commas because the [RFC 8259](https://tools.ietf.org/html/rfc8259) specification doesn't include them. For information about how to allow trailing commas, see [Allow comments and trailing commas](system-text-json-how-to.md#allow-comments-and-trailing-commas). There's no way to allow multiple trailing commas.
+
+## Character escaping
+
+During serialization, Newtonsoft is relatively permissive about letting characters through without escaping them. (That is, it doesn't replace them with `\uxxxx` where `xxxx` is the Unicode code of the character.) `System.Text.Json` escapes more characters by default to provide defense-in-depth protections against cross-site scripting (XSS) or information disclosure attacks. `System.Text.Json` escapes all non-ASCII characters by default, so you don't need to do anything if you're using `StringEscapeHandling.EscapeNonAscii`. For information about how to override the default `System.Text.Json` behavior, see [Customize character encoding](system-text-json-how-to.md#customize-character-encoding).
+
+## Converter registration precedence
+
+The Newtonsoft registration precedence for custom converters is as follows:
+
+* Attribute on property
+* Attribute on type
+* `Converters` collection
+
+This order means that a custom converter in the `Converters` collection is overridden by a converter that is registered by applying an attribute at the type level. Both of those registrations are overridden by an attribute at the property level.
+
+The `System.Text.Json` registration precedence for custom converters is different:
+
+* Attribute on property
+* `Converters` collection
+* Attribute on type
+
+The difference here is that a custom converter in the `Converters` collection overrides an attribute at the type level. The intention behind this order of precedence is to make run-time changes override design-time choices. There's no way to change the precedence.
+
+## Newtonsoft [JsonProperty] attribute
+
+Newtonsoft has a `[JsonProperty]` attribute that is used for a variety of functions. `System.Text.Json` has single-purpose attributes instead of a multi-purpose attribute. The following attributes serve purposes similar to some of the Newtonsoft attribute functions:
```

Comment:
Would it help to have a table of equivalency here? Something like this https://xunit.net/docs/comparisons would be really helpful and can evolve over time as new features are added.

Reply:
That's a great idea - I created a separate issue for it (#16309) and will address it in a separate PR.

Reply:
Another example is the table from `Newtonsoft.Json`, something like this:
https://www.newtonsoft.com/json
![image](https://user-images.githubusercontent.com/6527137/71871777-2710d700-30cf-11ea-8dba-764606533070.png)